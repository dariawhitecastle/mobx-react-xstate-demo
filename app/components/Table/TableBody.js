/* eslint-disable react/prop-types */
/**
 *
 * TableBody
 *
 */

import React from 'react'
import { TableBody } from 'grommet'
import { doc } from './TableBody.doc'
import helpers from '../../utils/helpers'

function createTableBody({ children, ...props }) {
  return <TableBody {...props}>{children}</TableBody>
}

export default helpers.createWithDoc({
  docFunction: doc,
  component: createTableBody,
})
