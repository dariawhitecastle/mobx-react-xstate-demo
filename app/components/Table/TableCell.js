/* eslint-disable react/prop-types */
/**
 *
 * TableCell
 *
 */

import React from 'react'
import { TableCell } from 'grommet'
import { doc } from './TableCell.doc'
import helpers from '../../utils/helpers'

function createTableCell({ children, plain, scope, size, verticalAlign, ...props }) {
  return (
    <TableCell plain={plain} scope={scope} size={size} verticalAlign={verticalAlign} {...props}>
      {children}
    </TableCell>
  )
}

export default helpers.createWithDoc({
  docFunction: doc,
  component: createTableCell,
})
