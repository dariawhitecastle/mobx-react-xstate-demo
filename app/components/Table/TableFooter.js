/* eslint-disable react/prop-types */
/**
 *
 * TableFooter
 *
 */

import React from 'react'
import { TableFooter } from 'grommet'
import { doc } from './TableFooter.doc'
import helpers from '../../utils/helpers'

function createTableFooter({ children, ...props }) {
  return <TableFooter {...props}>{children}</TableFooter>
}

export default helpers.createWithDoc({
  docFunction: doc,
  component: createTableFooter,
})
