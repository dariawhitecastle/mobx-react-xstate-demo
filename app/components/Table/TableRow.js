/* eslint-disable react/prop-types */
/**
 *
 * TableRow
 *
 */

import React from 'react'
import { TableRow } from 'grommet'
import { doc } from './TableRow.doc'
import helpers from '../../utils/helpers'

function createTableRow({ children, ...props }) {
  return <TableRow {...props}>{children}</TableRow>
}

export default helpers.createWithDoc({
  docFunction: doc,
  component: createTableRow,
})
