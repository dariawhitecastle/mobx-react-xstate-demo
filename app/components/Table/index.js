export { default as Table } from './Table'
export { default as TableBody } from './TableBody'
export { default as TableFooter } from './TableFooter'
export { default as TableHeader } from './TableHeader'
export { default as TableRow } from './TableRow'
export { default as TableCell } from './TableCell'
