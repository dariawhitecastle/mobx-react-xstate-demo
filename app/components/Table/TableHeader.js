/* eslint-disable react/prop-types */
/**
 *
 * TableHeader
 *
 */

import React from 'react'
import { TableHeader } from 'grommet'
import { doc } from './TableFooter.doc'
import helpers from '../../utils/helpers'

function createTableHeader({ children, ...props }) {
  return <TableHeader {...props}>{children}</TableHeader>
}

export default helpers.createWithDoc({
  docFunction: doc,
  component: createTableHeader,
})
