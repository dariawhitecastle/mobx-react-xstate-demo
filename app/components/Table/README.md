<!-- ! This is a generated file. To make changes, edit <Component>.doc.js ! -->
## createTable
A table of data organized in cells.

## Usage

```javascript
import { Table, TableHeader, TableFooter, TableBody, TableRow } from 'components/Table';
<Table />
```

## Properties

**a11yTitle**

Custom title to be used by screen readers.

```
string
```

**alignSelf**

How to align along the cross axis when contained in
      a Box or along the column axis when contained in a Grid.

```
start
center
end
stretch
```

**gridArea**

The name of the area to place
    this inside a parent Grid.

```
string
```

**margin**

The amount of margin around the component. An object can
      be specified to distinguish horizontal margin, vertical margin, and
      margin on a particular side.

```
none
xxsmall
xsmall
small
medium
large
xlarge
{
  bottom: 
    xxsmall
    xsmall
    small
    medium
    large
    xlarge
    string,
  horizontal: 
    xxsmall
    xsmall
    small
    medium
    large
    xlarge
    string,
  left: 
    xxsmall
    xsmall
    small
    medium
    large
    xlarge
    string,
  right: 
    xxsmall
    xsmall
    small
    medium
    large
    xlarge
    string,
  top: 
    xxsmall
    xsmall
    small
    medium
    large
    xlarge
    string,
  vertical: 
    xxsmall
    xsmall
    small
    medium
    large
    xlarge
    string
}
string
```

**caption**

One line description.

```
string
```
  
## Intrinsic element

```
table
```
## Theme
  
**global.edgeSize.responsiveBreakpoint**

The actual breakpoint to trigger changes in Table. Expects `string`.

Defaults to

```
small
```

**global.size**

The size that impacts max-width and width. Expects `undefined`.

Defaults to

```
{
      xxsmall: '48px',
      xsmall: '96px',
      small: '192px',
      medium: '384px',
      large: '768px',
      xlarge: '1152px',
      xxlarge: '1536px',
      full: '100%',
      }
```

**table.extend**

Any additional style for Table. Expects `string | (props) => {}`.

Defaults to

```
undefined
```
