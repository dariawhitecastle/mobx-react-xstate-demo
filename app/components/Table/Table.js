/* eslint-disable react/prop-types */
/**
 *
 * Table
 *
 */

import React from 'react'
import { Table } from 'grommet'
import { doc } from './Table.doc'
import helpers from '../../utils/helpers'

function createTable({ a11lyTitle, alignSelf, caption, children, margin, ...rest }) {
  return (
    <Table
      a11lyTitle={a11lyTitle}
      alignSelf={alignSelf}
      caption={caption}
      margin={margin}
      {...rest}
    >
      {children}
    </Table>
  )
}

export default helpers.createWithDoc({
  docFunction: doc,
  component: createTable,
})
