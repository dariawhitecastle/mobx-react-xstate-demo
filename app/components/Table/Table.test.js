/**
 *
 * Tests for Table
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react'
import { render } from '@testing-library/react'
// import 'jest-dom/extend-expect'; // add some helpful assertions

import Table from './Table'
import TableBody from './TableBody'
import TableHeader from './TableHeader'
import TableFooter from './TableFooter'
import TableRow from './TableRow'
import TableCell from './TableCell'
import { Header } from '../Header'

const columns = [
  {
    property: 'name',
    align: 'start',
    label: 'Name',
    dataScope: 'col',
  },
  {
    property: 'email',
    align: 'start',
    label: 'Email',
    dataScope: 'col',
  },
  {
    property: 'amount',
    align: 'start',
    label: 'Amount',
    dataScope: 'col',
  },
]

const data = [
  {
    id: 1,
    name: 'Eric',
    email: 'eric@gmail.com',
    amount: '$45',
  },
  {
    id: 2,
    name: 'Chris',
    email: 'chris@gmail.com',
    amount: '$45',
  },
  {
    id: 3,
    name: 'Shanna',
    email: 'shanna@gmail.com',
    amount: '$45',
  },
]

const renderComponent = (props = {}) =>
  render(
    <Table {...props}>
      <TableHeader>
        <TableRow>
          {columns.map(c => (
            <TableCell key={c.property} scope="col" align={c.align}>
              <Header level="4">{c.label}</Header>
            </TableCell>
          ))}
        </TableRow>
      </TableHeader>
      <TableBody>
        {data.map(datum => (
          <TableRow key={datum.id}>
            {columns.map(c => (
              <TableCell key={c.property} scope={c.dataScope} align={c.align}>
                <Header level="5">{c.format ? c.format(datum) : datum[c.property]}</Header>
              </TableCell>
            ))}
          </TableRow>
        ))}
      </TableBody>
      <TableFooter>
        <TableRow>
          {columns.map(c => (
            <TableCell key={c.property} align={c.align}>
              <Header level="4">{c.footer}</Header>
            </TableCell>
          ))}
        </TableRow>
      </TableFooter>
    </Table>,
  )

test('it renders and matches snapshot', () => {
  const { container } = renderComponent()
  expect(container).toMatchSnapshot()
})
