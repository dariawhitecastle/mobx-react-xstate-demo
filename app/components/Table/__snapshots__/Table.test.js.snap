// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders and matches snapshot 1`] = `
<div>
  <table
    class="StyledTable-sc-1m3u5g-6 fGUmP"
  >
    <thead
      class="StyledTable__StyledTableHeader-sc-1m3u5g-4 dHaGyd"
    >
      <tr
        class="StyledTable__StyledTableRow-sc-1m3u5g-2 kXgizX"
      >
        <th
          class="StyledTable__StyledTableCell-sc-1m3u5g-0 hNAgVv"
          scope="col"
        >
          <div
            class="StyledBox-sc-13pk1d4-0 eoKPJA"
          >
            <h4
              class="StyledHeading-sc-1rdh4aw-0 dOVRRX"
            >
              Name
            </h4>
          </div>
        </th>
        <th
          class="StyledTable__StyledTableCell-sc-1m3u5g-0 hNAgVv"
          scope="col"
        >
          <div
            class="StyledBox-sc-13pk1d4-0 eoKPJA"
          >
            <h4
              class="StyledHeading-sc-1rdh4aw-0 dOVRRX"
            >
              Email
            </h4>
          </div>
        </th>
        <th
          class="StyledTable__StyledTableCell-sc-1m3u5g-0 hNAgVv"
          scope="col"
        >
          <div
            class="StyledBox-sc-13pk1d4-0 eoKPJA"
          >
            <h4
              class="StyledHeading-sc-1rdh4aw-0 dOVRRX"
            >
              Amount
            </h4>
          </div>
        </th>
      </tr>
    </thead>
    <tbody
      class="StyledTable__StyledTableBody-sc-1m3u5g-3 dwInxx"
    >
      <tr
        class="StyledTable__StyledTableRow-sc-1m3u5g-2 kXgizX"
      >
        <th
          class="StyledTable__StyledTableCell-sc-1m3u5g-0 ezepRo"
          scope="col"
        >
          <div
            class="StyledBox-sc-13pk1d4-0 eAAZJM"
          >
            <h5
              class="StyledHeading-sc-1rdh4aw-0 jDZxiw"
            >
              Eric
            </h5>
          </div>
        </th>
        <th
          class="StyledTable__StyledTableCell-sc-1m3u5g-0 ezepRo"
          scope="col"
        >
          <div
            class="StyledBox-sc-13pk1d4-0 eAAZJM"
          >
            <h5
              class="StyledHeading-sc-1rdh4aw-0 jDZxiw"
            >
              eric@gmail.com
            </h5>
          </div>
        </th>
        <th
          class="StyledTable__StyledTableCell-sc-1m3u5g-0 ezepRo"
          scope="col"
        >
          <div
            class="StyledBox-sc-13pk1d4-0 eAAZJM"
          >
            <h5
              class="StyledHeading-sc-1rdh4aw-0 jDZxiw"
            >
              $45
            </h5>
          </div>
        </th>
      </tr>
      <tr
        class="StyledTable__StyledTableRow-sc-1m3u5g-2 kXgizX"
      >
        <th
          class="StyledTable__StyledTableCell-sc-1m3u5g-0 ezepRo"
          scope="col"
        >
          <div
            class="StyledBox-sc-13pk1d4-0 eAAZJM"
          >
            <h5
              class="StyledHeading-sc-1rdh4aw-0 jDZxiw"
            >
              Chris
            </h5>
          </div>
        </th>
        <th
          class="StyledTable__StyledTableCell-sc-1m3u5g-0 ezepRo"
          scope="col"
        >
          <div
            class="StyledBox-sc-13pk1d4-0 eAAZJM"
          >
            <h5
              class="StyledHeading-sc-1rdh4aw-0 jDZxiw"
            >
              chris@gmail.com
            </h5>
          </div>
        </th>
        <th
          class="StyledTable__StyledTableCell-sc-1m3u5g-0 ezepRo"
          scope="col"
        >
          <div
            class="StyledBox-sc-13pk1d4-0 eAAZJM"
          >
            <h5
              class="StyledHeading-sc-1rdh4aw-0 jDZxiw"
            >
              $45
            </h5>
          </div>
        </th>
      </tr>
      <tr
        class="StyledTable__StyledTableRow-sc-1m3u5g-2 kXgizX"
      >
        <th
          class="StyledTable__StyledTableCell-sc-1m3u5g-0 ezepRo"
          scope="col"
        >
          <div
            class="StyledBox-sc-13pk1d4-0 eAAZJM"
          >
            <h5
              class="StyledHeading-sc-1rdh4aw-0 jDZxiw"
            >
              Shanna
            </h5>
          </div>
        </th>
        <th
          class="StyledTable__StyledTableCell-sc-1m3u5g-0 ezepRo"
          scope="col"
        >
          <div
            class="StyledBox-sc-13pk1d4-0 eAAZJM"
          >
            <h5
              class="StyledHeading-sc-1rdh4aw-0 jDZxiw"
            >
              shanna@gmail.com
            </h5>
          </div>
        </th>
        <th
          class="StyledTable__StyledTableCell-sc-1m3u5g-0 ezepRo"
          scope="col"
        >
          <div
            class="StyledBox-sc-13pk1d4-0 eAAZJM"
          >
            <h5
              class="StyledHeading-sc-1rdh4aw-0 jDZxiw"
            >
              $45
            </h5>
          </div>
        </th>
      </tr>
    </tbody>
    <tfoot
      class="StyledTable__StyledTableFooter-sc-1m3u5g-5 zMIVX"
    >
      <tr
        class="StyledTable__StyledTableRow-sc-1m3u5g-2 kXgizX"
      >
        <td
          class="StyledTable__StyledTableCell-sc-1m3u5g-0 edcjKh"
        >
          <div
            class="StyledBox-sc-13pk1d4-0 bGwdLS"
          >
            <h4
              class="StyledHeading-sc-1rdh4aw-0 dOVRRX"
            />
          </div>
        </td>
        <td
          class="StyledTable__StyledTableCell-sc-1m3u5g-0 edcjKh"
        >
          <div
            class="StyledBox-sc-13pk1d4-0 bGwdLS"
          >
            <h4
              class="StyledHeading-sc-1rdh4aw-0 dOVRRX"
            />
          </div>
        </td>
        <td
          class="StyledTable__StyledTableCell-sc-1m3u5g-0 edcjKh"
        >
          <div
            class="StyledBox-sc-13pk1d4-0 bGwdLS"
          >
            <h4
              class="StyledHeading-sc-1rdh4aw-0 dOVRRX"
            />
          </div>
        </td>
      </tr>
    </tfoot>
  </table>
</div>
`;
